.circleci/config.yml

```diff
@@ -53,8 +57,97 @@ jobs:
       - run: make test
       - *save_cache
 
+  push-docker-pr:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - setup_remote_docker
+    - run: make build-image
+    - run: ./scripts/push-docker-pr.sh
+    - run:
+        name: Re-run Docker Push if fail
+        command: |
+            sleep 20
+            ./scripts/push-docker-pr.sh
```

Comment:
Are we absolutely sure that consecutive docker pushes are idempotent on failure? I am a bit concerned if some image gets half pushed, and then on retry, gets into an even weird state.

Reply:
@agnivade The image should not reach the dockerhun unless its fully pushed. This trick used here is to solve one issue we were seeing a lot in the cloud as well, where circleci was failing mostly because of networking issues and at least we have a retry in place for that.

Reply:
Sounds good.